File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-5.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    void dfs(TreeNode* node, int& ret, int mx) {
        mx = max(mx, node->val);
        // cout << node->val << " " << mx << endl;
        if (node->left != nullptr) {
            if (node->left->val >= mx) {
                ret++;
            }
            dfs(node->left, ret, mx);
        }
        if (node->right != nullptr) {
            if (node->right->val >= mx) {
                ret++;
            }
            dfs(node->right, ret, mx);
        }
        
    }
    int goodNodes(TreeNode* root) {
        int ret {1};
        dfs(root, ret, INT_MIN);
        return ret;
    }
};

File: Data Structures & Algorithms/longest-palindromic-substring/submission-5.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        maxString = ''
        def dfs(r, l):
            if (l < 0) or (r >= len(s)):
                return s[l + 1: r]
            if (s[r] != s[l]): 
                return s[l + 1: r]

            return dfs(r +1 , l - 1)
        
        for i in range(len(s)):
            maxString = max(maxString, dfs(i, i), dfs(i + 1, i), key=len)
        return maxString